### PR TITLE
Add git root inside a git repo

### PR DESCRIPTION
**Pull request by gazorby**
## WHAT
Inside a git repository (either at root or under sub directories), the list is shifted towards the bottom so the first element become the git root folder of the repository.

Support has been added for both zsh and fish.
## WHY
#110

### CODE REVIEW

**Comment by b4b4r07** [comment]
Wow, amazing. I'll check it later

**Comment by b4b4r07** [comment]
@Gazorby 

Sorry for the late response and thank you for the contribution!


Could you fix the conflicts?

**Comment by hoshsadiq** [comment]
Slight request for improvement, can we change it so that it doesn't show if you're at the git root dir?

File: src/entry.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+__enhancd::entry::git::root()
+{
+    if git rev-parse --is-inside-work-tree 2> /dev/null && [[ $(git rev-parse --show-toplevel) != $PWD ]]; then
```
Comment (hoshsadiq) [inline src/entry.sh:3]: I suggest to quote the `$(git rev-parse --show-toplevel)` and `$PWD` to prevent issues with spaces in dir/file names

**Comment by hoshsadiq** [comment]
Thanks for the update @Gazorby!

File: src/entry.sh (hunk 1 above)
Comment (Gazorby) [reply src/entry.sh:3]: According to the [bash manual](https://www.gnu.org/software/bash/manual/bash.html#Conditional-Constructs), 

> Word splitting and filename expansion are not performed on the words between the [[ and ]]

So i don't think it's necessary

Comment (hoshsadiq) [reply src/entry.sh:3]: Cool! Didn't know that.

**Comment by hoshsadiq** [comment]
Any update on this?

**Comment by jorgebucaran** [comment]
@hoshsadiq We should probably merge #139 first and then come back to this.

**Pull request merged**